### PR TITLE
Capture visibility events

### PR DESCRIPTION
**Pull request by mhennoch**
Captured as 0 duration spans as these can't be complete spans(due to us not being able to close some of them in some case) and we don't have events. Disabled for now cause we need to make sure that sending them is fine. At some point we probably need to add this info to each span so data can be filtered in UI.

Code in index.ts is moved cause otherwise some things fire at wrong times.

### CODE REVIEW

File: src/SplunkPageVisibilityInstrumentation.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+export class SplunkPageVisibilityInstrumentation extends InstrumentationBase {
+  unloading: boolean;
+
+  constructor(config: InstrumentationConfig = {}) {
+    super(MODULE_NAME, VERSION, Object.assign({}, config));
+    this.unloading = false;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
```
Comment (jtmal-signalfx) [inline src/SplunkPageVisibilityInstrumentation.ts:33]: Can this be configured in eslint? Especially, if all instances of `InstrumentationBase` require `init()` to be present.

File: src/SplunkPageVisibilityInstrumentation.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { hrTime } from '@opentelemetry/core';
+import { InstrumentationBase, InstrumentationConfig } from '@opentelemetry/instrumentation';
+import { VERSION } from './version';
+
+const MODULE_NAME = 'splunk-visibility';
+let visibilityListener: any;
```
Comment (jtmal-signalfx) [inline src/SplunkPageVisibilityInstrumentation.ts:22]: Can these listeners be fields of the instrumentation?

File: integration-tests/utils/globals.js [hunk 3]
```diff
@@ -86,11 +86,12 @@ module.exports = {
...
     Object.assign(browser.globals, await buildBackendContext(browser));
 
-    browser.globals.emulateTabSwitchingAway = async () => {
-      await browser.execute(() => {
-        Object.defineProperty(document, 'visibilityState', { value: 'hidden', configurable: true });
-        document.dispatchEvent(new Event('visibilitychange'));
-      });
+    browser.globals.emulateTabSwitching = async (visible) => {
+      await browser.execute((hidden) => {
+        Object.defineProperty(document, 'hidden', { value: hidden, configurable: true });
```
Comment (jtmal-signalfx) [inline integration-tests/utils/globals.js:91]: Looking at this, I'd suggest we make it a unit test. This isn't how an integration test should work.

**Review by jtmal-signalfx — APPROVED** [review]
No major issues, I'd prefer that the tests for visibility are unit tests because of how they work.

File: src/SplunkPageVisibilityInstrumentation.ts (hunk 1 above)
Comment (mhennoch) [reply src/SplunkPageVisibilityInstrumentation.ts:33]: Switched off globally

File: src/SplunkPageVisibilityInstrumentation.ts (hunk 2 above)
Comment (mhennoch) [reply src/SplunkPageVisibilityInstrumentation.ts:22]: ✅

File: integration-tests/utils/globals.js (hunk 3 above)
Comment (mhennoch) [reply integration-tests/utils/globals.js:91]: Will look into making it a real integration test with a real tab switch at some point

**Review by t2t2 — APPROVED** [review]

**Pull request merged**